sound/soc/sof/topology.c

```diff
@@ -3005,6 +3053,7 @@ static void sof_complete(struct snd_soc_component *scomp)
 			break;
 		}
 	}
+	snd_sof_get_kcontrol_val(sdev);
```

Comment:
I am still lost on how the controls are initialized
1. by topology
2. by reading DSP values over IPC?
3. by alsa-mixer?

Reply:
@plbossart Currently, we don't have the ability to get the default value by topology. So it is by reading DSP value over IPC for now. And it may be overwrite by alsa-mixer after booting up.

Reply:
ok, you should add a comment on this so that the sequence is documented.

Reply:
@plbossart done